### DISCUSSION TITLE
Changes I made to simplemenu configuration upon upgrading (some of which might be worth considering in the next version?)

### DISCUSSION THREAD

**Discussion by valzi** [eduardofilo/RG350_adam_image#187] [Ideas & Requests]
I moved SGB to be next to GB and GBC, in the handhelds category.

I removed .bin from the romexts for cd-based consoles. This prevents duplicate, non-functional entries in Simplemenu.

I deleted all quake and quake 2 stuff because it caused Simplemenu sections I did not want. The rg280v does not have suitable controls for playing Quake (though it's acceptable for Doom.) That said, I would probably have appreciated having it ready to go on one of the other supported consoles. It's less work to remove it then to add it, so maybe inclusion is better.

I moved /media/data/apps/regba_rg350_odbeta.opk to the front of the GBA execs list and I moved /media/data/apps/PocketSNES_new.opk to the front of the SNES execs list. There is a strong consensus in the community that they are generally much better options than the Retroarch options. There are certainly reasons to prefer using Retroarch, of course, but I would argue that's not a good default for these two consoles.

**Comment by neilswann80** [comment]
I'd say these are all personal preferences and there are arguments for either option, but for simplicity (and less additional work for @eduardofilo) leaving things as they are and concentrating on the seamless integration that Retroarch provides is the best course from a multi-handheld image like Adam.

The SGB category is an interesting dilemma.  Whilst they are primarily handheld games, although some have been modified with the SGB in mind and therefore have the option to use the greater hardware specs, they are played on a console.  Do we keep the Gameboy games themselves all in the same section, or remain consistent with the differentiation between handheld and console?  For my personal config I'll be leaning towards the latter and leaving as-is (for now).

**Reply by neilswann80** [reply]
@eduardofilo 
For Quake/Quake2 you could however modify the "S10_create_ext_scaffolding.sh" to only create the .fbl shortcuts (and therefore only show the Quake menus) if the .pak files are present.
I've attached a modified bash script that works on my RG350m.
[S10_create_ext_scaffolding.sh.txt](https://github.com/eduardofilo/RG350_adam_image/files/8039893/S10_create_ext_scaffolding.sh.txt)

**Reply by eduardofilo** [reply]
Yes sir. That's a good approximation to the solution. If you want, make me a PR to the repository to incorporate it as your contribution.
Many thanks.

**Reply by valzi** [reply]
Neat!

**Comment by neilswann80** [comment]
@valzi
FYI... the change I suggested regarding the Quake/Quake2 shortcuts was approved.  So in future versions the shortcuts and therefore the menus will only appear for Quake/Quake2 if you have populated their ROM directories with pak files; essentially they should only be visible if you intend on playing them (this was the default scenario before the shortcut fix was applied to allow for "friendly" naming of the individual Quake games; now we have the best of both!).

**Reply by eduardofilo** [reply]
Thank you @neilswann80.
Great contribution.

**Reply by valzi** [reply]
Thanks, that's great!
